core/src/main/java/bio/terra/pearl/core/service/notification/email/AdminEmailService.java

```diff
@@ -4,45 +4,63 @@
+import bio.terra.pearl.core.model.study.Study;
+import bio.terra.pearl.core.model.study.StudyEnvironment;
 import bio.terra.pearl.core.service.notification.NotificationContextInfo;
 import bio.terra.pearl.core.service.notification.substitutors.AdminEmailSubstitutor;
+import bio.terra.pearl.core.service.portal.PortalEnvironmentService;
+import bio.terra.pearl.core.service.rule.EnrolleeContext;
+import bio.terra.pearl.core.service.study.StudyEnvironmentService;
+import bio.terra.pearl.core.service.study.StudyService;
 import bio.terra.pearl.core.shared.ApplicationRoutingPaths;
 import com.sendgrid.Mail;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.text.StringSubstitutor;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.Nullable;
+import java.util.UUID;
+
 @Service
 @Slf4j
 public class AdminEmailService {
   private static final String WELCOME_EMAIL_TEMPLATE_STABLEID = "broad_admin_welcome";
   private static final int WELCOME_EMAIL_TEMPLATE_VERSION = 1;
-  private EmailTemplateService emailTemplateService;
-  private ApplicationRoutingPaths routingPaths;
-  private SendgridClient sendgridClient;
-
-
+  private final StudyEnvironmentService studyEnvironmentService;
+  private final StudyService studyService;
+  private final PortalEnvironmentService portalEnvironmentService;
+  private final EmailTemplateService emailTemplateService;
+  private final ApplicationRoutingPaths routingPaths;
+  private final SendgridClient sendgridClient;
 
-  public AdminEmailService( EmailTemplateService emailTemplateService,
-                           ApplicationRoutingPaths routingPaths, SendgridClient sendgridClient) {
+  public AdminEmailService(EmailTemplateService emailTemplateService,
+                           ApplicationRoutingPaths routingPaths,
+                           SendgridClient sendgridClient,
+                           StudyEnvironmentService studyEnvironmentService,
+                           StudyService studyService,
+                           PortalEnvironmentService portalEnvironmentService) {
     this.emailTemplateService = emailTemplateService;
     this.routingPaths = routingPaths;
     this.sendgridClient = sendgridClient;
+    this.studyEnvironmentService = studyEnvironmentService;
+    this.studyService = studyService;
+    this.portalEnvironmentService = portalEnvironmentService;
   }
 
   @Async
   public void sendWelcomeEmail(Portal portal, AdminUser adminUser) {
     NotificationContextInfo contextInfo = loadContextInfo(WELCOME_EMAIL_TEMPLATE_STABLEID, WELCOME_EMAIL_TEMPLATE_VERSION, portal);
-    sendEmail(contextInfo, adminUser);
+    sendEmail(contextInfo, adminUser, null);
   }
 
-  public void sendEmail(NotificationContextInfo contextInfo, AdminUser adminUser) {
+  public void sendEmail(NotificationContextInfo contextInfo, AdminUser adminUser, @Nullable EnrolleeContext enrolleeContext) {
```

Comment:
Our current practices disfavor `@Nullable` annotations.  Instead, use a comment to explain what will happen if it's null
```suggestion
  public void sendEmail(NotificationContextInfo contextInfo, AdminUser adminUser, EnrolleeContext enrolleeContext) {
```